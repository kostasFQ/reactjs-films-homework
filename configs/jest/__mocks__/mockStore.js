export const movie = {
  queryString: '',
  startFetch: false,
  finishFetch: true,
  trailerWindow: false,
  currentUrl: 'https://api.themoviedb.org/3/movie/now_playing?api_key=3a2531498f834486708efbfa60ac046b&language=en-US&page=1',
  film: {
    vote_count: 1576,
    id: 335983,
    video: false,
    vote_average: 6.6,
    title: 'Venom',
    popularity: 371.566,
    poster_path: '/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
    original_language: 'en',
    original_title: 'Venom',
    genre_ids: [
      878,
      28,
      80,
      28,
      27
    ],
    backdrop_path: '/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg',
    adult: false,
    overview: 'When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.',
    release_date: '2018-10-03'
  },
  fullResponse: [
    {
      vote_count: 1576,
      id: 335983,
      video: false,
      vote_average: 6.6,
      title: 'Venom',
      popularity: 371.566,
      poster_path: '/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      original_language: 'en',
      original_title: 'Venom',
      genre_ids: [
        878,
        28,
        80,
        28,
        27
      ],
      backdrop_path: '/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg',
      adult: false,
      overview: 'When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.',
      release_date: '2018-10-03'
    },
    {
      vote_count: 192,
      id: 424139,
      video: false,
      vote_average: 6.7,
      title: 'Halloween',
      popularity: 209.869,
      poster_path: '/lNkDYKmrVem1J0aAfCnQlJOCKnT.jpg',
      original_language: 'en',
      original_title: 'Halloween',
      genre_ids: [
        27
      ],
      backdrop_path: '/hO1oTBGNxO5fBKVEuWnSpICJH7c.jpg',
      adult: false,
      overview: 'Laurie Strode comes to her final confrontation with Michael Myers, the masked figure who has haunted her since she narrowly escaped his killing spree on Halloween night four decades ago.',
      release_date: '2018-10-18'
    },
    {
      vote_count: 1003,
      id: 332562,
      video: false,
      vote_average: 7.5,
      title: 'A Star Is Born',
      popularity: 192.779,
      poster_path: '/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg',
      original_language: 'en',
      original_title: 'A Star Is Born',
      genre_ids: [
        18,
        10402,
        10749
      ],
      backdrop_path: '/840rbblaLc4SVxm8gF3DNdJ0YAE.jpg',
      adult: false,
      overview: 'Seasoned musician Jackson Maine discovers—and falls in love with—struggling artist Ally. She has just about given up on her dream to make it big as a singer—until Jack coaxes her into the spotlight. But even as Ally\'s career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.',
      release_date: '2018-10-03'
    },
    {
      vote_count: 3018,
      id: 260513,
      video: false,
      vote_average: 7.6,
      title: 'Incredibles 2',
      popularity: 147.981,
      poster_path: '/x1txcDXkcM65gl7w20PwYSxAYah.jpg',
      original_language: 'en',
      original_title: 'Incredibles 2',
      genre_ids: [
        28,
        12,
        16,
        10751
      ],
      backdrop_path: '/mabuNsGJgRuCTuGqjFkWe1xdu19.jpg',
      adult: false,
      overview: 'Elastigirl springs into action to save the day, while Mr. Incredible faces his greatest challenge yet – taking care of the problems of his three children.',
      release_date: '2018-06-14'
    },
    {
      vote_count: 154,
      id: 347375,
      video: false,
      vote_average: 5.7,
      title: 'Mile 22',
      popularity: 143.679,
      poster_path: '/ptSrT1JwZFWGhjSpYUtJaasQrh.jpg',
      original_language: 'en',
      original_title: 'Mile 22',
      genre_ids: [
        28
      ],
      backdrop_path: '/eTwftrInxzZTSLUkX5hoifOczKJ.jpg',
      adult: false,
      overview: 'A CIA field officer and an Indonesian police officer are forced to work together in confronting political corruption. An informant must be moved twenty-two miles to safety.',
      release_date: '2018-08-16'
    },
    {
      vote_count: 292,
      id: 369972,
      video: false,
      vote_average: 7.3,
      title: 'First Man',
      popularity: 117.407,
      poster_path: '/i91mfvFcPPlaegcbOyjGgiWfZzh.jpg',
      original_language: 'en',
      original_title: 'First Man',
      genre_ids: [
        36,
        18
      ],
      backdrop_path: '/z6ZANhPCcPl1HsKGjY5iZALf65V.jpg',
      adult: false,
      overview: 'A look at the life of the astronaut, Neil Armstrong, and the legendary space mission that led him to become the first man to walk on the Moon on July 20, 1969.',
      release_date: '2018-10-11'
    },
    {
      vote_count: 651,
      id: 346910,
      video: false,
      vote_average: 5.3,
      title: 'The Predator',
      popularity: 108.934,
      poster_path: '/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg',
      original_language: 'en',
      original_title: 'The Predator',
      genre_ids: [
        27,
        878,
        28,
        53
      ],
      backdrop_path: '/f4E0ocYeToEuXvczZv6QArrMDJ.jpg',
      adult: false,
      overview: 'From the outer reaches of space to the small-town streets of suburbia, the hunt comes home. Now, the universe’s most lethal hunters are stronger, smarter and deadlier than ever before, having genetically upgraded themselves with DNA from other species. When a young boy accidentally triggers their return to Earth, only a ragtag crew of ex-soldiers and a disgruntled science teacher can prevent the end of the human race.',
      release_date: '2018-09-13'
    },
    {
      vote_count: 225,
      id: 463272,
      video: false,
      vote_average: 6.2,
      title: 'Johnny English Strikes Again',
      popularity: 97.249,
      poster_path: '/tCBxnZwLiY1BOKw3tH6AxHZdqPh.jpg',
      original_language: 'en',
      original_title: 'Johnny English Strikes Again',
      genre_ids: [
        12,
        10751,
        28,
        35
      ],
      backdrop_path: '/yCOLqh5MOGyYdo58Ap0aWvKop9h.jpg',
      adult: false,
      overview: 'Disaster strikes when a criminal mastermind reveals the identities of all active undercover agents in Britain. The secret service can now rely on only one man—Johnny English. Currently teaching at a minor prep school, Johnny springs back into action to find the mysterious hacker. For this mission to succeed, he’ll need all of his skills—what few he has—as the man with yesterday’s analogue methods faces off against tomorrow’s digital technology.',
      release_date: '2018-09-13'
    },
    {
      vote_count: 1191,
      id: 439079,
      video: false,
      vote_average: 5.8,
      title: 'The Nun',
      popularity: 95.278,
      poster_path: '/sFC1ElvoKGdHJIWRpNB3xWJ9lJA.jpg',
      original_language: 'en',
      original_title: 'The Nun',
      genre_ids: [
        27,
        9648,
        53
      ],
      backdrop_path: '/fgsHxz21B27hOOqQBiw9L6yWcM7.jpg',
      adult: false,
      overview: 'When a young nun at a cloistered abbey in Romania takes her own life, a priest with a haunted past and a novitiate on the threshold of her final vows are sent by the Vatican to investigate. Together they uncover the order’s unholy secret. Risking not only their lives but their faith and their very souls, they confront a malevolent force in the form of the same demonic nun that first terrorized audiences in “The Conjuring 2,” as the abbey becomes a horrific battleground between the living and the damned.',
      release_date: '2018-09-05'
    },
    {
      vote_count: 705,
      id: 487558,
      video: false,
      vote_average: 7.5,
      title: 'BlacKkKlansman',
      popularity: 88.868,
      poster_path: '/bT5WuAsjDJYQv2vXbWGDQTmjKav.jpg',
      original_language: 'en',
      original_title: 'BlacKkKlansman',
      genre_ids: [
        80,
        18,
        35,
        36
      ],
      backdrop_path: '/gMVdhfQ7q9DFHhDkehrququjGPd.jpg',
      adult: false,
      overview: 'Ron Stallworth, an African-American police officer from Colorado, successfully manages to infiltrate the local Ku Klux Klan and become the head of the local chapter.',
      release_date: '2018-08-09'
    },
    {
      vote_count: 310,
      id: 399360,
      video: false,
      vote_average: 5.3,
      title: 'Alpha',
      popularity: 77.414,
      poster_path: '/afdZAIcAQscziqVtsEoh2PwsYTW.jpg',
      original_language: 'en',
      original_title: 'Alpha',
      genre_ids: [
        12,
        18
      ],
      backdrop_path: '/nKMeTdm72LQ756Eq20uTjF1zDXu.jpg',
      adult: false,
      overview: 'After a hunting expedition goes awry, a young caveman struggles against the elements to find his way home.',
      release_date: '2018-08-17'
    },
    {
      vote_count: 45,
      id: 442062,
      video: false,
      vote_average: 5.8,
      title: 'Goosebumps 2: Haunted Halloween',
      popularity: 60.913,
      poster_path: '/8bcpki9GfXdXj9esFpPtlate8v0.jpg',
      original_language: 'en',
      original_title: 'Goosebumps 2: Haunted Halloween',
      genre_ids: [
        12,
        35,
        14,
        27,
        10751
      ],
      backdrop_path: '/9xYg9ScjGaO2HrLfOfYZirTrB3o.jpg',
      adult: false,
      overview: 'Two boys face an onslaught from witches, monsters, ghouls and a talking dummy after they discover a mysterious book by author R.\u202fL. Stine.',
      release_date: '2018-10-11'
    },
    {
      vote_count: 8,
      id: 523873,
      video: false,
      vote_average: 4.1,
      title: 'Kung Fu League',
      popularity: 58.776,
      poster_path: '/rW0A73hjzPWVwADlCTLnjLhAFLX.jpg',
      original_language: 'zh',
      original_title: '功夫联盟',
      genre_ids: [
        28,
        35
      ],
      backdrop_path: null,
      adult: false,
      overview: 'Martial arts comedy following a group of kung fu legends banding together to take on the bad guys. The legends includes VINCENT ZHAO reprising his role as ‘Wong Fei Hung’ with DENNIS TO once again portraying ‘Wing Chun’ master ‘Ip Man’, DANNY CHAN KWOK KWAN as ‘Chen Zhen’ and ANDY ON as master ‘Huo Yuan Jia’.',
      release_date: '2018-10-19'
    },
    {
      vote_count: 95,
      id: 454293,
      video: false,
      vote_average: 6.1,
      title: 'Night School',
      popularity: 57.828,
      poster_path: '/1NSMAaBPSbWv7sGmF8oLGMNb8Qm.jpg',
      original_language: 'en',
      original_title: 'Night School',
      genre_ids: [
        35
      ],
      backdrop_path: '/7CWpvqVY5JWbEtkaenzqfJjVSal.jpg',
      adult: false,
      overview: 'Teddy Walker is a successful salesman whose life takes an unexpected turn when he accidentally blows up his place of employment. Forced to attend night school to get his GED, Teddy soon finds himself dealing with a group of misfit students, his former high school nemesis and a feisty teacher who doesn\'t think he\'s too bright.',
      release_date: '2018-09-27'
    },
    {
      vote_count: 124,
      id: 446894,
      video: false,
      vote_average: 6.4,
      title: 'Smallfoot',
      popularity: 57.483,
      poster_path: '/4nKoB6wMVXfsYgRZK5lHZ5VMQ6J.jpg',
      original_language: 'en',
      original_title: 'Smallfoot',
      genre_ids: [
        35,
        16,
        10751,
        12,
        14
      ],
      backdrop_path: '/7t88SoT3Dd8DhGnQuVoSbMNUl3W.jpg',
      adult: false,
      overview: 'A bright young yeti finds something he thought didn\'t exist—a human. News of this “smallfoot” throws the simple yeti community into an uproar over what else might be out there in the big world beyond their snowy village.',
      release_date: '2018-09-20'
    },
    {
      vote_count: 183,
      id: 460885,
      video: false,
      vote_average: 6.4,
      title: 'Mandy',
      popularity: 55.079,
      poster_path: '/m0yf7J7HsKeK6E81SMRcX8vx6mH.jpg',
      original_language: 'en',
      original_title: 'Mandy',
      genre_ids: [
        28,
        53,
        27
      ],
      backdrop_path: '/tNa19CK0CQZl5rxZ35QRdKAT2s0.jpg',
      adult: false,
      overview: 'The Shadow Mountains, 1983. Red and Mandy lead a loving and peaceful existence; but when their pine-scented haven is savagely destroyed, Red is catapulted into a phantasmagoric journey filled with bloody vengeance and laced with fire.',
      release_date: '2018-09-13'
    },
    {
      vote_count: 18,
      id: 424694,
      video: false,
      vote_average: 8.4,
      title: 'Bohemian Rhapsody',
      popularity: 50.787,
      poster_path: '/jf7jiy0rDVn1ljUwv3pWKousQ1p.jpg',
      original_language: 'en',
      original_title: 'Bohemian Rhapsody',
      genre_ids: [
        18,
        10402
      ],
      backdrop_path: '/jNUpYq2jRSwQM89vST9yQZelMSu.jpg',
      adult: false,
      overview: 'Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock \'n\' roll band Queen in 1970. Hit songs become instant classics. When Mercury\'s increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet – finding a way to keep the band together amid the success and excess.',
      release_date: '2018-10-24'
    },
    {
      vote_count: 1090,
      id: 447200,
      video: false,
      vote_average: 6.1,
      title: 'Skyscraper',
      popularity: 49.399,
      poster_path: '/5LYSsOPzuP13201qSzMjNxi8FxN.jpg',
      original_language: 'en',
      original_title: 'Skyscraper',
      genre_ids: [
        28,
        53,
        18
      ],
      backdrop_path: '/oMKFQmoVgB69fyXfSMu0lGlHJP2.jpg',
      adult: false,
      overview: 'Framed and on the run, a former FBI agent must save his family from a blazing fire in the world\'s tallest building.',
      release_date: '2018-07-11'
    },
    {
      vote_count: 111,
      id: 446021,
      video: false,
      vote_average: 6.5,
      title: 'Bad Times at the El Royale',
      popularity: 48.039,
      poster_path: '/iNtFgXqXPRMkm1QO8CHn5sHfUgE.jpg',
      original_language: 'en',
      original_title: 'Bad Times at the El Royale',
      genre_ids: [
        53,
        80,
        9648
      ],
      backdrop_path: '/gb3TVVZNNxVGNfS1NxGaiWZfwnW.jpg',
      adult: false,
      overview: 'Seven strangers, each with a secret to bury, meet at Lake Tahoe\'s El Royale, a rundown hotel with a dark past in 1969. Over the course of one fateful night, everyone will have a last shot at redemption.',
      release_date: '2018-10-04'
    },
    {
      vote_count: 886,
      id: 400155,
      video: false,
      vote_average: 6.8,
      title: 'Hotel Transylvania 3: Summer Vacation',
      popularity: 47.376,
      poster_path: '/gjAFM4xhA5vyLxxKMz38ujlUfDL.jpg',
      original_language: 'en',
      original_title: 'Hotel Transylvania 3: Summer Vacation',
      genre_ids: [
        10751,
        14,
        35,
        16
      ],
      backdrop_path: '/m03jul0YdVEOFXEQVUv6pOVQYGL.jpg',
      adult: false,
      overview: 'Dracula, Mavis, Johnny and the rest of the Drac Pack take a vacation on a luxury Monster Cruise Ship, where Dracula falls in love with the ship’s captain, Ericka, who’s secretly a descendant of Abraham Van Helsing, the notorious monster slayer.',
      release_date: '2018-06-28'
    }
  ]
}