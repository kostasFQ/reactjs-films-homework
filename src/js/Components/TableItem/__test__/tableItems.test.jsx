import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { movie } from '../../../../../configs/jest/__mocks__/mockStore';
import ConnectedTableItem from '../index';

const initialState = { movie };
const mStore = configureMockStore([thunk]);
const store = mStore(initialState);
const ITEM = {
  vote_count: 1613,
  id: 335983,
  video: false,
  vote_average: 6.6,
  title: 'Venom',
  popularity: 376.826,
  poster_path: '/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
  original_language: 'en',
  original_title: 'Venom',
  genre_ids: [
    878,
    28,
    80,
    28,
    27,
  ],
  backdrop_path: '/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg',
  adult: false,
  overview: 'When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.',
  release_date: '2018-10-03',
};


test('1.test TABLE_ITEM component render', () => {
  const item = { ...ITEM };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const result = testRenderer.toJSON();
  expect(result.type).toBe('div');
  expect(result).toMatchSnapshot();
});

test('6.test TABLE_ITEM component render', () => {
  const item = { ...ITEM };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const testInstance = testRenderer.root;

  const infoBut = testInstance.findByProps({ name: 'View Info' });
  infoBut.props.action();
  const but = testInstance.findByProps({ name: 'Watch Now' });
  but.props.action();
  const result = testRenderer.toJSON();
  expect(result).toMatchSnapshot();
});

test('2.test TABLE_ITEM component render', () => {
  const item = { ...ITEM, poster_path: undefined };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const result = testRenderer.toJSON();
  expect(result.type).toBe('div');
  expect(result).toMatchSnapshot();
});

test('3.test TABLE_ITEM component render', () => {
  const item = { ...ITEM, visiblePreview: true };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const testInstance = testRenderer.root;
  const but = testInstance.findAllByType('button');
  but[0].props.onClick();
  but[1].props.onClick();
  const result = testRenderer.toJSON();
  expect(result).toMatchSnapshot();
});

test('4.test TABLE_ITEM component render', () => {
  const item = { ...ITEM, visiblePreview: false };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const testInstance = testRenderer.root;
  const but = testInstance.findByProps({ name: 'Watch Now' });
  but.props.action();
  const result = testRenderer.toJSON();
  expect(result).toMatchSnapshot();
});

test('5.test TABLE_ITEM component render', () => {
  const item = { ...ITEM, poster_path: undefined };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><ConnectedTableItem {...item} /></Provider>,
  );
  const testInstance = testRenderer.root;
  const div = testInstance.find(el => el.type === 'div' && el.props.className === 'info');
  div.props.onMouseLeave();
  const result = testRenderer.toJSON();
  expect(result).toMatchSnapshot();
});
