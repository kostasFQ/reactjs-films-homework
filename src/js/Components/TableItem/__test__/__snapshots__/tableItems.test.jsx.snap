// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://image.tmdb.org/t/p/w300//2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      width="100%"
    />
  </div>
  <div
    className="description"
  >
    <div
      className="descBox"
    >
      <span
        className="title"
      >
        VENOM
      </span>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
    <div
      className="cut"
    >
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Action
        </li>
        <li
          className="item"
        >
          Crime
        </li>
        <li
          className="item"
        >
          Horror
        </li>
        <li
          className="item"
        >
          Science Fiction
        </li>
      </ul>
    </div>
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="buttonBlock"
        >
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="play"
            />
          </button>
          <div
            className="name"
          >
            Watch Now
          </div>
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            View Info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://wpmovies.scriptburn.com/wp-content/themes/wp_movies/images/noposter.jpg"
      width="100%"
    />
  </div>
  <div
    className="description"
  >
    <div
      className="descBox"
    >
      <span
        className="title"
      >
        VENOM
      </span>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
    <div
      className="cut"
    >
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Action
        </li>
        <li
          className="item"
        >
          Crime
        </li>
        <li
          className="item"
        >
          Horror
        </li>
        <li
          className="item"
        >
          Science Fiction
        </li>
      </ul>
    </div>
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="buttonBlock"
        >
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="play"
            />
          </button>
          <div
            className="name"
          >
            Watch Now
          </div>
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            View Info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`3.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://image.tmdb.org/t/p/w300//2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      width="100%"
    />
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="innerDescr"
        >
          <div
            className="up"
          >
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              âœ•
            </button>
          </div>
          <div
            className="descBox"
          >
            <span
              className="hoverTitle"
            >
              VENOM
            </span>
            <div
              className="container"
            >
              6.6
            </div>
          </div>
          <div
            className="cut"
          >
            <ul
              className="list"
            >
              <li
                className="item"
              >
                Action
              </li>
              <li
                className="item"
              >
                Crime
              </li>
              <li
                className="item"
              >
                Horror
              </li>
              <li
                className="item"
              >
                Science Fiction
              </li>
            </ul>
          </div>
          <div
            className="overview"
          >
            <div
              className="container"
            >
              <div
                className="buttonsBlock"
              >
                <div>
                  When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "marginLeft": "-50px",
                "width": "200%",
              }
            }
          />
          <div
            className="w"
          >
            <button
              className="container"
              onClick={[Function]}
              type="button"
            >
              Watch Now
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`4.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://image.tmdb.org/t/p/w300//2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      width="100%"
    />
  </div>
  <div
    className="description"
  >
    <div
      className="descBox"
    >
      <span
        className="title"
      >
        VENOM
      </span>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
    <div
      className="cut"
    >
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Action
        </li>
        <li
          className="item"
        >
          Crime
        </li>
        <li
          className="item"
        >
          Horror
        </li>
        <li
          className="item"
        >
          Science Fiction
        </li>
      </ul>
    </div>
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="buttonBlock"
        >
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="play"
            />
          </button>
          <div
            className="name"
          >
            Watch Now
          </div>
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            View Info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`5.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://wpmovies.scriptburn.com/wp-content/themes/wp_movies/images/noposter.jpg"
      width="100%"
    />
  </div>
  <div
    className="description"
  >
    <div
      className="descBox"
    >
      <span
        className="title"
      >
        VENOM
      </span>
      <div
        className="container"
      >
        6.6
      </div>
    </div>
    <div
      className="cut"
    >
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Action
        </li>
        <li
          className="item"
        >
          Crime
        </li>
        <li
          className="item"
        >
          Horror
        </li>
        <li
          className="item"
        >
          Science Fiction
        </li>
      </ul>
    </div>
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="buttonBlock"
        >
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            <div
              className="play"
            />
          </button>
          <div
            className="name"
          >
            Watch Now
          </div>
          <button
            className="container"
            onClick={[Function]}
            type="button"
          >
            View Info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`6.test TABLE_ITEM component render 1`] = `
<div
  className="container"
>
  <div
    className="miniPoster"
  >
    <img
      alt="poster"
      height="100%"
      src="http://image.tmdb.org/t/p/w300//2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      width="100%"
    />
  </div>
  <div
    className="info"
    onMouseLeave={[Function]}
  >
    <div />
    <div
      className="previewField"
    >
      <div
        className="inner"
      >
        <div
          className="innerDescr"
        >
          <div
            className="up"
          >
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              âœ•
            </button>
          </div>
          <div
            className="descBox"
          >
            <span
              className="hoverTitle"
            >
              VENOM
            </span>
            <div
              className="container"
            >
              6.6
            </div>
          </div>
          <div
            className="cut"
          >
            <ul
              className="list"
            >
              <li
                className="item"
              >
                Action
              </li>
              <li
                className="item"
              >
                Crime
              </li>
              <li
                className="item"
              >
                Horror
              </li>
              <li
                className="item"
              >
                Science Fiction
              </li>
            </ul>
          </div>
          <div
            className="overview"
          >
            <div
              className="container"
            >
              <div
                className="buttonsBlock"
              >
                <div>
                  When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "marginLeft": "-50px",
                "width": "200%",
              }
            }
          />
          <div
            className="w"
          >
            <button
              className="container"
              onClick={[Function]}
              type="button"
            >
              Watch Now
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
